server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java

```diff
@@ -113,10 +113,8 @@ private static void checkSupportedVersion(IndexMetadata indexMetadata, IndexVers
                     + indexMetadata.getCompatibilityVersion()
                     + "] but the minimum compatible version is ["
                     + minimumIndexCompatibilityVersion
-                    + "]. It should be re-indexed in Elasticsearch "
-                    + (Version.CURRENT.major - 1)
-                    + ".x before upgrading to "
-                    + Version.CURRENT
+                    + "]. It should be re-indexed using the previous major version of Elasticsearch before upgrading to "
```

Comment:
@thecoop I rephrased this to avoid an explicit version; do you think it might work or better to revert and just use majors here?

Reply:
I'm a little reluctant to add a semantic concept like a major version in Build.current(). What will serverless return? Or maybe you are saying that 8 will be the "perpetual major" (the final ES release) and we should never increment it again?

Reply:
On the other hand, for this case (index compatibility) maybe it makes sense to just leave Version.major: maybe for this kind of things we will always have a concept of compatibility through major versions, and it is fine as-is.

Reply:
I don't expect 8 to be the perpetual major. My thought was actually that I don't know what we will be doing about major versions, but we could at least centralize any logic for them so that we can make our changes in one place.

What we need to convey here is that if a user is upgrading a node to 8.11 but has an index from 6.x or earlier, they need to first upgrade to 7.x (preferably 7.17.x), and _then_ upgrade to 8.x.

In settings that we're managing, we could probably understand something like "a node that supports org.apache.lucene.util.Version.CURRENT.major - 1", but if I recall correctly the Lucene major and Elasticsearch major are not aligned. That would be confusing for a lot of self-managing users.

I don't really know what to do, but I'd be fine to leave Version.major here for now and come back to it in a later PR.

Reply:
I agree with you, we probably need a better way to express this information or concept. It may be linked to the "era" concept or features that Simon is working on (e.g. 7.x could be mapped to a `can-reindex-6x`). Also, it could be that moving to IndexVersion changes the definition of index compatibility.
I will revert my change and leave Version.major here, making  a note to  revisit it when we have a clearer direction.
Thanks!